core/src/main/java/org/mule/api/lifecycle/LifecycleUtils.java

```diff
@@ -60,31 +93,81 @@ public static void startIfNeeded(Object object) throws MuleException
         }
     }
 
+    /**
+     * For each item in the {@code objects} collection, it checks if it implements
+     * the {@link Startable} interface and if so it invokes the {@link Startable#start()}
+     * method on it
+     *
+     * @param objects the list of objects to be started
+     * @throws MuleException
+     */
     public static void startIfNeeded(Collection<? extends Object> objects) throws MuleException
     {
         doApplyPhase(Startable.PHASE_NAME, objects, null);
     }
 
+    /**
+     * For each item in the {@code objects} collection, it checks if it implements
+     * the {@link Stoppable} interface and if so it invokes the {@link Stoppable#stop()}
+     * method on it
+     *
+     * @param objects the list of objects to be stopped
+     * @throws MuleException
+     */
     public static void stopIfNeeded(Collection<? extends Object> objects) throws MuleException
     {
         doApplyPhase(Stoppable.PHASE_NAME, objects, null);
     }
 
+    /**
+     * if {@code object} implements the {@link Stoppable} interface,
+     * then it invokes the {@link Stoppable#stop()} method on it.
+     *
+     * @param object the object you're trying to stop
+     * @throws MuleException
+     */
     public static void stopIfNeeded(Object object) throws MuleException
     {
         doApplyPhase(Stoppable.PHASE_NAME, Arrays.asList(object), null);
     }
 
+    /**
+     * if {@code object} implements the {@link Disposable} interface,
+     * then it invokes the {@link Disposable#dispose()} method on it.
+     * If the dispose operation fails, then the exception will be silently
+     * logged using this class {@link #LOGGER}
+     *
+     * @param object the object you're trying to dispose
+     */
     public static void disposeIfNeeded(Object object)
```

Comment:
Is not clear to me why there are scenarios where I would like to log into this class logger instead of the app logger